### ISSUE TITLE
"hanoi" demo doesn't work

### ISSUE THREAD

**Issue by martinwguy** [martinwguy/xvi#40]
The "[hanoi solving vi macros](https://github.com/martinwguy/xvi/files/649652/hanoi.txt)" don't work under xvi.
Under classic vi and vim.tiny, the initialization macro, "I", gives:
```
01234567
0
0


T0123456n
T0123456$
/
```
with one initial blank line, then the "run" macro ("L") whizzes until it shows
```
0
01234567
0


T0123456n
T0123456$
/
```
with one initial blank line, and stops.
(nvi does the same but without the leading blank line because the result of the initialization macro, "I", is wrong).

xvi looks like it gets the right macro definitions, as shown by ":map" and ":map!", and the initialization ("I") works but when running ("L") it whizzes forever in an uninterruptable infinite loop, showing
```
07
065432107


T0123456n
T0123456$
```
with two initial blank lines and one trailing blank line and with the 7's flickering on and off.

**Comment by martinwguy** [comment]
The first deviation of xvi's behaviour is in the "L" macro
```
map L 1G/t^MX/^0^M$P1GJ$An$BGC0e$X0E0F$X/T^M@f^M@h^M$A1GJ@f0l$Xn$PU
```
at the character C, mapped to "fp (put named buffer f). classic vi give "/6" while xvi says "Nothing to put!" on the status line and appends nothing. Named buffer f should have been set by the A character, mapped to "fyl

**Comment by martinwguy** [comment]
"yl" at end of line is fixed by 4871502b

Now the first visible difference is when doing the first "@f".
At this point, in classic vi, buffer f contains "/6" but in xvi it contains "0" (zero).

**Comment by martinwguy** [comment]
"y2l" on penultimate char of line is fixed by 8753db40

Now it seems to solve the problem but fails to stop when it has done so.

**Comment by martinwguy** [comment]
Fixed by 7cbc6673. Hanoi now works.
